src/test/java/org/opentripplanner/middleware/ApiUserFlowTest.java

```diff
@@ -196,12 +197,21 @@ public void canSimulateApiUserFlow() throws URISyntaxException, JsonProcessingEx
 
         // Create a monitored trip for an Otp user authenticating as an Api user. An Api user can create a monitored
         // trip for an Otp user they created.
+
+        // Set mock OTP responses so that trip existence checks in the
+        // POST call below to save the monitored trip can pass.
+        TestUtils.setupOtpMocks(ItineraryUtilsTest.createMockOtpResponsesForTripExistence());
+
         HttpResponse<String> createTripResponseAsApiUser = makeRequest(
             MONITORED_TRIP_PATH,
             JsonUtils.toJson(monitoredTrip),
             apiUserHeaders,
             HttpUtils.REQUEST_METHOD.POST
         );
+
+        // After POST is complete, reset mock OTP responses for subsequent tests.
+        TestUtils.resetOtpMocks();
+
```

Comment:
These should be moved outside of this test into before/after test methods, especially the resetOtpMocks call. If the `makeRequest` call fails, the OTP mocks won't be reset.

Reply:
I tried moving the reset call in https://github.com/ibi-group/otp-middleware/pull/102/commits/98f91c7d7043e0039a39a1855dc3e40befbf178e and that caused failures, because there are other mock OTP calls that happen before the test exits, so I'm resetting mocks here (see https://github.com/ibi-group/otp-middleware/pull/102/commits/53ef5e41fdf5e4022d3c585e414c05f8f078d9f9 and https://github.com/ibi-group/otp-middleware/pull/102/commits/51f7dbcff06174eb82ac377d7baae340b0c9f6d5) in addition to resetting inside `@AfterEach`.